VersionControlUI/src/nodes/DiffComparisonPair.h

```diff
@@ -63,38 +56,42 @@ class VERSIONCONTROLUI_API DiffComparisonPair : public Super<Visualization::UINo
 			Model::Text* newVersionObjectPath();
 			Model::Text* oldVersionObjectPath();
 
+			Model::Text* singleObjectPath();
+
 			Model::Text* componentType();
 
+			bool twoObjectPathsDefined();
+
 
 		private:
+			bool twoObjectPathsDefined_;
+
 			Model::Node* oldVersionNode_{};
 			Model::Node* newVersionNode_{};
 
 			Model::Text* newVersionObjectPath_{};
 			Model::Text* oldVersionObjectPath_{};
 
+			Model::Text* singleObjectPath_{};
 
 			Model::Text* componentType_{};
 
+			QString computeObjectPath(Model::Node* node);
 
+			void setUpComponentType();
```

Comment:
this name could be confused with the name of a setter. change `setUp` to `compute`, and the same for the method below.